common/src/main/java/org/opensearch/ml/common/controller/MLRateLimiter.java

```diff
@@ -0,0 +1,141 @@
+        this.rateLimitNumber = in.readOptionalString();
+        if (in.readBoolean()) {
+            this.rateLimitUnit = in.readEnum(TimeUnit.class);
+        }
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeOptionalString(rateLimitNumber);
+        if (rateLimitUnit != null) {
+            out.writeBoolean(true);
+            out.writeEnum(rateLimitUnit);
+        } else {
+            out.writeBoolean(false);
+        }
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params params) throws IOException {
+        builder.startObject();
+        if (rateLimitNumber != null) {
+            builder.field(RATE_LIMIT_NUMBER_FIELD, rateLimitNumber);
+        }
+        if (rateLimitUnit != null) {
+            builder.field(RATE_LIMIT_UNIT_FIELD, rateLimitUnit);
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    public void update(MLRateLimiter updateContent) {
+        if (updateContent.getRateLimitNumber() != null) {
+            this.rateLimitNumber = updateContent.getRateLimitNumber();
+        }
+        if (updateContent.getRateLimitUnit() != null) {
+            this.rateLimitUnit = updateContent.getRateLimitUnit();
+        }
+    }
+
+    public static MLRateLimiter update(MLRateLimiter rateLimiter, MLRateLimiter updateContent) {
+        if (rateLimiter == null) {
+            return updateContent;
+        } else {
+            rateLimiter.update(updateContent);
+            return rateLimiter;
+        }
+    }
+
+    public static boolean isUpdatable(MLRateLimiter rateLimiter, MLRateLimiter updateContent) {
+        if (updateContent == null) {
+            return false;
+        } else if (rateLimiter == null) {
+            return true;
+        } else if (updateContent.isRateLimiterEmpty()) {
+            return false;
+        } else return (!Objects.equals(updateContent.getRateLimitNumber(), rateLimiter.getRateLimitNumber()) && updateContent.getRateLimitNumber() != null)
+                || (!Objects.equals(updateContent.getRateLimitUnit(), rateLimiter.getRateLimitUnit()) &&  updateContent.getRateLimitUnit() != null);
+    }
+
+    public static boolean isDeployRequiredAfterUpdate(MLRateLimiter rateLimiter, MLRateLimiter updateContent) {
+        if (!isUpdatable(rateLimiter, updateContent)) {
+            return false;
+        } else {
+            return rateLimiter.isRateLimiterConstructable() || updateContent.isRateLimiterConstructable()
+                    || (rateLimiter.getRateLimitUnit() != null && updateContent.getRateLimitNumber() != null)
+                    || (rateLimiter.getRateLimitNumber() != null && updateContent.getRateLimitUnit() != null);
+        }
+    }
+
+    public boolean isRateLimiterConstructable() {
+        return (this.rateLimitUnit != null && this.rateLimitNumber != null);
+    }
+
+    public boolean isRateLimiterEmpty() {
```

Comment:
We don't need so verbose name
```suggestion
    public boolean isEmpty() {
```

Reply:
Changed